### PR TITLE
feat(wallet) Add NFT Pinning

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/26828

Sec review  : https://github.com/brave/security/issues/1182
## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### 1. First Time NFT Pinning Flow:
https://user-images.githubusercontent.com/48117473/216613044-581aab1c-197f-4175-8ad0-0f7daa9edba6.mov

### 2. NFT Pinning for a newly added token
https://user-images.githubusercontent.com/48117473/216613368-baaaceb0-1535-48a6-9854-457b3736c2fd.mov

### CODE REVIEW

**Comment by yrliou** [comment]
@muliswilliam @cypt4 just a friendly note that nothing should be accessible/triggered without enabling the feature flag, and please open a security & privacy issue to prepare for enabling the feature flag linking to this PR and the core PR, thanks.

File: components/brave_wallet_ui/common/hooks/nft-pin.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      const pinnable = nfts
+        .filter(({ canBePinned }) => canBePinned)
+        .map(({ token }) => token)
+      setNonFungibleTokens(nfts)
+      setPinnableNfts(pinnable)
+    }
+
+    getTokensSupport()
+      .catch(console.error)
+  }, [isNftPinningFeatureEnabled, userVisibleTokensInfo])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/common/hooks/nft-pin.ts:98]: Missing `isTokenPinningSupported` dependancy.

File: components/brave_wallet_ui/common/hooks/nft-pin.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/common/hooks/nft-pin.ts:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/inspect-nfts/components/info-tooltip/info-tooltip.style.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/components/info-tooltip/info-tooltip.style.ts:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/inspect-nfts/components/info-tooltip/info-tooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/components/info-tooltip/info-tooltip.tsx:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/inspect-nfts/components/nft-list/nft-list.styles.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/components/nft-list/nft-list.styles.ts:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/inspect-nfts/components/nft-list/nft-list.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/components/nft-list/nft-list.tsx:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/inspect-nfts/inspect-nfts.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/inspect-nfts.tsx:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/inspect-nfts/inspect-nfts.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+export const InspectNftsScreen = ({ onClose }: Props) => {
+  const [showTooltip, setShowTooltip] = React.useState<boolean>(false)
+
+  // routing
+  const history = useHistory()
+
+  // redux
+  const dispatch = useDispatch()
+  const isAutoPinEnabled = useSafePageSelector(PageSelectors.isAutoPinEnabled)
+
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/inspect-nfts.tsx:62]: Can we add a `// methods` comment?

File: components/brave_wallet_ui/components/desktop/inspect-nfts/inspects-nfts.styles.ts [hunk 9]
```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/inspects-nfts.styles.ts:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/inspect-nfts/inspects-nfts.styles.ts [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+
+export const InfoSubHeading = styled.h6`
+  font-family: 'Poppins';
+  font-style: normal;
+  font-weight: 600;
+  font-size: 12px;
+  line-height: 18px;
+  margin: 0;
+  padding: 0;
+  color: #AEB1C2;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/inspects-nfts.styles.ts:112]: Do we have a `theme` variable for this? if not should we add a `(prefers-color-scheme: dark)` here?

File: components/brave_wallet_ui/components/desktop/inspect-nfts/inspects-nfts.styles.ts [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+  height: 14px;
+  -webkit-mask-image: url(${Info});
+  mask-image: url(${Info});
+  cursor: pointer;
+`
+
+export const SubDivider = styled.div`
+  width: 100%;
+  height: 1px;
+  background-color: rgba(233, 233, 244, 0.3);
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/inspects-nfts.styles.ts:127]: Do we have a `theme` variable for this? if not should we add a `(prefers-color-scheme: dark)` here?

File: components/brave_wallet_ui/components/desktop/inspect-nfts/components/info-tooltip/info-tooltip.style.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+  margin: 0;
+  color: #F6F6FA;
+`
+
+export const ArrowUp = styled.div`
+  width: 0; 
+  height: 0; 
+  border-left: 8px solid transparent;
+  border-right: 8px solid transparent;
+  border-bottom: 8px solid #1E2029;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/components/info-tooltip/info-tooltip.style.ts:46]: Do we have a `theme` variable for this? if not should we add a `(prefers-color-scheme: dark)` here?

File: components/brave_wallet_ui/components/desktop/inspect-nfts/components/info-tooltip/info-tooltip.style.ts [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+import styled from 'styled-components'
+
+export const StyledWrapper = styled.div`
+  position: absolute;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  z-index: 3;
+  background-color: #1E2029;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/inspect-nfts/components/info-tooltip/info-tooltip.style.ts:14]: Do we have a `theme` variable for this? if not should we add a `(prefers-color-scheme: dark)` here?

File: components/brave_wallet_ui/components/desktop/local-ipfs-node/local-ipfs-node.styles.ts [hunk 14]
```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/local-ipfs-node/local-ipfs-node.styles.ts:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/local-ipfs-node/local-ipfs-node.tsx [hunk 15]
```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/local-ipfs-node/local-ipfs-node.tsx:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/local-ipfs-node/local-ipfs-node.tsx [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+export const LocalIpfsNodeScreen = (props: Props) => {
+  const { onClose } = props
+
+  // routing
+  const history = useHistory()
+
+  // redux
+  const dispatch = useDispatch()
+  const isAutoPinEnabled = useSafePageSelector(PageSelectors.isAutoPinEnabled)
+
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/local-ipfs-node/local-ipfs-node.tsx:61]: Can we add a `// methods` comment here?

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts [hunk 17]
```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.tsx [hunk 18]
```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.tsx:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.tsx [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+import {
+  StyledWrapper,
+  Text,
+  LearnMore,
+  CloseButton
+} from './nft-ipfs-banner.style'
+
+export const NftIpfsBanner = ({ onDismiss }: Props) => {
+  const history = useHistory()
+
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.tsx:40]: Can we add a `// redux` comment here?

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.tsx [hunk 20]
```diff
@@ -0,0 +1,91 @@
...
+  const { pinnableNftsCount, pinnedNftsCount, pinningStatusSummary: status } = useNftPin()
+  const isAutoPinEnabled = useSafePageSelector(PageSelectors.isAutoPinEnabled)
+
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.tsx:43]: Can we add a `// memos` comment here?

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.tsx [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+    switch (status) {
+      case BraveWallet.TokenPinStatusCode.STATUS_PINNED:
+        return 'success'
+      case BraveWallet.TokenPinStatusCode.STATUS_PINNING_IN_PROGRESS:
+        return 'uploading'
+      default:
+        return 'success'
+    }
+  }, [status, pinnableNftsCount, isAutoPinEnabled])
+
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.tsx:56]: Can we add a `// methods` comment here?

File: components/brave_wallet_ui/components/desktop/nft-pinning-status-animation/nft-pinning-status-animation.style.ts [hunk 22]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status-animation/nft-pinning-status-animation.style.ts:4]: Capitalize `You`

File: components/brave_wallet_ui/components/desktop/nft-pinning-status-animation/nft-pinning-status-animation.tsx [hunk 23]
```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status-animation/nft-pinning-status-animation.tsx:4]: Capitalize `You`

File: components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts [hunk 24]
```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts [hunk 25]
```diff
@@ -0,0 +1,169 @@
...
+const getBackground = (p: BraveThemedStyledProps<any>, status: BraveWallet.TokenPinStatusCode) => {
+  switch (status) {
+    case BraveWallet.TokenPinStatusCode.STATUS_PINNING_IN_PROGRESS:
+      return '#FFFCF0'
+
+    case BraveWallet.TokenPinStatusCode.STATUS_PINNING_FAILED:
+      return '#FFF0F2'
+
+    case BraveWallet.TokenPinStatusCode.STATUS_PINNED:
+      return 'rgba(213, 245, 218, 1)'
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts:23]: Do these need `dark-mode` theming?

File: components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts [hunk 26]
```diff
@@ -0,0 +1,169 @@
...
+  justify-content: center;
+  width: 10px;
+  height: 10px;
+  mask-repeat: no-repeat;
+`
+
+export const UploadIcon = styled(StatusIcon)`
+  mask-image: url(${Upload});
+  -webkit-mask-image: url(${Upload});
+  background-color: #FEBF17;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts:69]: Same here?

File: components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts [hunk 27]
```diff
@@ -0,0 +1,169 @@
...
+export const UploadIcon = styled(StatusIcon)`
+  mask-image: url(${Upload});
+  -webkit-mask-image: url(${Upload});
+  background-color: #FEBF17;  
+`
+
+export const CloseIcon = styled(StatusIcon)`
+  mask-image: url(${Close});
+  -webkit-mask-image: url(${Close});
+  background-color: #E32444;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts:75]: Same here?

File: components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts [hunk 28]
```diff
@@ -0,0 +1,169 @@
...
+  -webkit-mask-image: url(${Close});
+  background-color: #E32444;  
+`
+
+export const CheckIcon = styled(StatusIcon)`
+  width: 14px;
+  height: 14px;
+  mask-image: url(${Check});
+  -webkit-mask-image: url(${Check});
+  background-color: #4DC661;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.style.ts:83]: Same here?

File: components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.tsx [hunk 29]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.tsx:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.tsx [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+  CheckIcon
+} from './nft-pinning-status.style'
+
+interface Props {
+  pinningStatusCode: BraveWallet.TokenPinStatusCode
+}
+
+export const NftPinningStatus = (props: Props) => {
+  const { pinningStatusCode } = props
+  const [icon, setIcon] = React.useState<React.ReactNode>()
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.tsx:24]: `// state`

File: components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.tsx [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+  const [message, setmessage] = React.useState<string>('')
+
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-pinning-status/nft-pinning-status.tsx:26]: `// effects`

File: components/brave_wallet_ui/components/desktop/views/crypto/index.tsx [hunk 32]
```diff
@@ -134,6 +148,18 @@ const CryptoView = (props: Props) => {
...
+    history.push(WalletRoutes.Nfts)
+  }, [])
+
+  const onBack = React.useCallback(() => {
+    if (location.key) {
+      history.goBack()
+    } else {
+      history.push(WalletRoutes.Nfts)
+    }
+  }, [])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/views/crypto/index.tsx:161]: Missing `location.key` dependency.

File: components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.styles.tsx [hunk 33]
```diff
@@ -38,3 +41,31 @@ export const EmptyStateText = styled.div`
...
+  width: 24px;
+  height: 24px;
+  mask-image: url(${Ipfs});
+  -webkit-mask-image: url(${Ipfs});
+  mask-repeat: no-repeat;
+  background-color: #495057;
+  cursor: pointer;
+  
+  &:hover {
+    background-color: #469EA2;
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.styles.tsx:69]: does this need `dark-mode` theming?

File: components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx [hunk 34]
```diff
@@ -58,6 +68,10 @@ export const Nfts = (props: Props) => {
     dispatch(WalletPageActions.updateNFTMetadata(undefined))
   }, [dispatch])
 
+  const onClickIpfsButton = React.useCallback(() => {
+    onToggleShowIpfsBanner()
+  }, [])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx:73]: Missing `onToggleShowIpfsBanner` dependency.

File: components/brave_wallet_ui/components/desktop/views/nfts/nft-view.tsx [hunk 35]
```diff
@@ -7,33 +7,46 @@ import * as React from 'react'
...
+      )
     }
 
-    return userVisibleTokensInfo.filter(token => token.chainId === selectedNetworkFilter.chainId && (token.isErc721 || token.isNft))
+    return userVisibleTokensInfo.filter(
+      (token) =>
+        token.chainId === selectedNetworkFilter.chainId &&
+        (token.isErc721 || token.isNft)
+    )
   }, [userVisibleTokensInfo, selectedNetworkFilter])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/views/nfts/nft-view.tsx:47]: `selectednetworkFilter.chaindId`

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/ipfs-node-status/ipfs-node-status.style.ts [hunk 36]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/ipfs-node-status/ipfs-node-status.style.ts:4]: Update to 2023 and capitalize You

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/ipfs-node-status/ipfs-node-status.tsx [hunk 37]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/ipfs-node-status/ipfs-node-status.tsx:4]: Update to 2023 and capitalize You

File: components/brave_wallet_ui/components/shared/nft-icon/nft-icon-with-network-icon.tsx [hunk 38]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/shared/nft-icon/nft-icon-with-network-icon.tsx:4]: Update to `2023` and capitalize `You`

File: components/brave_wallet_ui/nft/components/nft-details/nft-details-styles.ts [hunk 39]
```diff
@@ -220,3 +221,62 @@ export const ErrorMessage = styled.p`
...
   margin-top: 50px;
 `
+
+export const NftStandard = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  padding: 2px 4px;
+  gap: 4px;
+  border: 1px solid rgba(174, 177, 194, 0.55);
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/nft/components/nft-details/nft-details-styles.ts:231]: Does this need `dark-mode` theming?

**Review by Douglashdaniel — CHANGES_REQUESTED** [review]

**Review by Douglashdaniel — APPROVED** [review]
Looks good to me!
Nice work! 🔥

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 40]
```diff
@@ -59,16 +59,16 @@ interface Props {
...
-  networks,
-  renderToken,
-  hideAddButton,
-  enableScroll,
-  maxListHeight,
-  hideAssetFilter,
-  hideAccountFilter,
-  hideAutoDiscovery
-}: Props) => {
+                             userAssetList,
```
Comment (cypt4) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:62]: Do we need this changes?

**Review by cypt4 — APPROVED** [review]
LGTM

**Review by stoletheminerals — APPROVED** [review]
It's ok to be merged since it's behind a feature flag. The review is still in process, so we shouldn't enable the feature by default

**Pull request merged**